src/actions/hid.action.ts

```diff
@@ -356,6 +356,39 @@ export const hidActionsThunk = {
     dispatch(KeymapActions.clearSelectedPos());
     dispatch(HeaderActions.updateFlashing(false));
   },
+
+  fetchSwitchMatrixState: (): ThunkPromiseAction<void> => async (
+    dispatch: ThunkDispatch<RootState, undefined, ActionTypes>,
+    getState: () => RootState
+  ) => {
+    const { app, entities } = getState();
+    const keyboard: IKeyboard = entities.keyboard!;
+
+    const result = await keyboard.fetchSwitchMatrixState();
+    if (!result.success) {
+      console.error(result.cause);
+      dispatch(NotificationActions.addError(result.error!, result.cause));
+      return;
+    }
+
+    const rows = entities.keyboardDefinition!.matrix.rows;
+    const cols = entities.keyboardDefinition!.matrix.cols;
+    const state = result.state!;
+
+    let i = 0;
+    for (let row = 0; row < rows; row++) {
+      let rowState = 0;
+      for (let colIdx = 0; colIdx < Math.ceil(cols/8); colIdx++) {
+        rowState = (rowState << 8) | state[i++];
+      }
+
+      for (let col = 0; col < cols; col++) {
+        if ((rowState & (1 << col)) !== 0) {
+          dispatch(AppActions.addTestedMatrix(row + ',' + col));
```

Comment:
@rtitmuss To handle key release events and to clear highlighting of each key top, you need to call the interface to clear them. @adamrocker will create the new interface. So could you call it after implementing of the new interface?

Reply:
Yes, I'll make the changes to support the new interface.

Reply:
@rtitmuss adamrocker created the new interface at: https://github.com/remap-keys/remap/blob/feat-347-test-matrix/src/components/configure/keymapToolbar/KeymapToolbar.container.ts#L41

You can set the `updateCurrentTestMatrix()` function with an argument. The argument is an array of "Pressed down key position at that time". The `addTestedMatrix()` function is used to highlight the key top. In the other hand, the `updateCurrentTestMatrix()` function is used for representing key press/release state. Could you use the `updateCurrentTestMatrix()` function?

Reply:
I've subumitted an update.

Reply:
Ok, I will test it.